src/main/java/com/epam/bookstoreservice/controller/BookStoreController.java

```diff
@@ -22,113 +23,124 @@
+
 /**
  * controller for bookstore
  */
 @RestController
 @RequestMapping("/v1/bookstore")
 @AllArgsConstructor
-@Api
+@Tag(name = "bookstore")
 public class BookStoreController {
 
     private final BookstoreService bookstoreService;
 
-    @PostMapping("/add-new-book")
-    @ApiOperation("add new book")
-    @ApiImplicitParams({ @ApiImplicitParam(paramType = "header",defaultValue = "Bearer ",dataType = "String", name = "Authorization", required = true) })
-    public ResponseEntity<BookResponseDTO> addNewBook(BookRequestDTO book) {
+    private final BookResponseDTOAssembler bookInfoAssembler;
 
+    @PostMapping("/add-new-book")
+    @Operation(description = "add new book")
+    public ResponseEntity<BookModel> addNewBook(BookRequestDTO bookRequestDTO) {
         return ResponseEntity
-                .status(HttpStatus.OK).body(bookstoreService.addNewBook(book));
-
+                .status(HttpStatus.OK)
+                .body(bookInfoAssembler.toModel(bookstoreService.addNewBook(bookRequestDTO)));
     }
 
-
     @PostMapping("/add-book")
-    @ApiOperation("add existed book")
-    @ApiImplicitParams({ @ApiImplicitParam(paramType = "header",defaultValue = "Bearer ",dataType = "String", name = "Authorization", required = true) })
-    public ResponseEntity<BookResponseDTO> addExistentBook(BookRequestDTO book) {
-
+    @Operation(description = "add existed book")
+    public ResponseEntity<BookModel> addExistentBook(BookRequestDTO bookRequestDTO) {
         return ResponseEntity
-                .status(HttpStatus.OK).body(bookstoreService.addExistentBook(book));
+                .status(HttpStatus.OK)
+                .body(bookInfoAssembler.toModel(bookstoreService.addExistentBook(bookRequestDTO)));
     }
 
     @GetMapping("/book/{id}")
-    @ApiOperation("get book by id")
-    @ApiImplicitParams({ @ApiImplicitParam(paramType = "header",defaultValue = "Bearer ",dataType = "String", name = "Authorization", required = true) })
-    public ResponseEntity<BookResponseDTO> getBookById(@PathVariable Integer id) {
-
+    @Operation(description = "get book by id")
+    public ResponseEntity<BookModel> getBookById(@PathVariable Integer id) {
         return ResponseEntity
-                .status(HttpStatus.OK).body(bookstoreService.getBookById(id));
-
+                .status(HttpStatus.OK)
+                .body(bookInfoAssembler.toModel(bookstoreService.getBookById(id)));
     }
 
     @GetMapping("/book-list")
-    @ApiOperation("get all books")
-    @ApiImplicitParams({ @ApiImplicitParam(paramType = "header",defaultValue = "Bearer ",dataType = "String", name = "Authorization", required = true) })
-    public ResponseEntity<List<BookResponseDTO>> getAllBooks() {
-
+    @Operation(description = "get all books")
+    public ResponseEntity<CollectionModel<BookModel>> getAllBooks() {
+        CollectionModel<BookModel> bookModels = bookInfoAssembler.toCollectionModel(bookstoreService.getAllBooks());
```

Comment:
Try to override toCollectionModel method in assembler  to simplified this method